jina/peapods/peas/__init__.py

```diff
@@ -123,6 +124,8 @@ def terminate(self):
         """
         if hasattr(self.worker, 'terminate'):
             self.worker.terminate()
+            # This sleep gives the process some time to terminate, otherwise we encountered hanging joins
+            time.sleep(0.1)
```

Comment:
i'm okay with adding sleep. but just to notice that, as Pea lives in the main process, this wait slows down the terminating process and makes it noticable sync. In a Flow with 10 Peas (quite common in even hello world), this means the terminating will take at least 1s.

By theory, the shutting down of this Pea and that Pea should be completely independent and can be done in parallel. So no matter how many Peas I have, closing them should just be a constant time, not linear time.

Reply:
terminating only happens when start did not work. So it is quite an abnormal behavior

Reply:
actually we came to the conclusion that we dont need the terminate at all? The runtime loop gets interrupted and will eventually exit and the process will die. Why call terminate on it then? I am checking locally if I still the error if removed